core/appeal/service.go

```diff
@@ -225,8 +221,15 @@ func (s *Service) Create(ctx context.Context, appeals []*domain.Appeal, opts ...
 			}
 		}
 
-		if err := s.checkExtensionEligibility(appeal, provider, policy, activeGrants); err != nil {
-			return fmt.Errorf("checking grant extension eligibility: %w", err)
+		activeGrant, err := s.findActiveGrant(ctx, appeal)
+		if err != nil {
+			return err
+		}
+
+		if activeGrant != nil {
```

Comment:
is it the expected behavior if `activeGrant` is nil (no active grant found), the flow is still valid?

Reply:
yes, the flow should not changed other than the query, we have the existing test cases to cover this 
https://github.com/odpf/guardian/blob/main/core/appeal/service_test.go